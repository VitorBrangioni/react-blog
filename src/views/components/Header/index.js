import React, { Component } from "react";
// import './scriptMenu';
import "./styles.css";
import { Header } from "./styles";

class HeaderComponent extends Component {

  constructor() {
    super();
    this.state = {
      show: false
    }
  }

  render() {
    const { show } = this.state;

    return (
      <Header>
        <div className={`menu-section container ${!show || 'on' }`}>
          <img
            src="https://blog.axur.com/hs-fs/hubfs/Ativo%204.png?width=454&name=Ativo%204.png%20454w,%20https://blog.axur.com/hs-fs/hubfs/Ativo%204.png?width=907&name=Ativo%204.png%20907w,%20https://blog.axur.com/hs-fs/hubfs/Ativo%204.png?width=1361&name=Ativo%204.png%201361w,%20https://blog.axur.com/hs-fs/hubfs/Ativo%204.png?width=1814&name=Ativo%204.png%201814w,%20https://blog.axur.com/hs-fs/hubfs/Ativo%204.png?width=2268&name=Ativo%204.png%202268w,%20https://blog.axur.com/hs-fs/hubfs/Ativo%204.png?width=2721&name=Ativo%204.png%202721w"
            alt="test"
          />
          <div className="menu-toggle" onClick={this.handleToggleMenu.bind(this)}>
            <div className="one"></div>
            <div className="two"></div>
            <div className="three"></div>
          </div>
        </div>
      </Header>
    );
  }

  handleToggleMenu() {
    const { show } = this.state;
    this.setState({ show: !show });
    document.body.style.overflow = !show ? 'hidden' : 'initial';
  }
}

export default HeaderComponent;
