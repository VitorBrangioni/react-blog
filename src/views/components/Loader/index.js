import React from 'react';
import ContentLoader from "react-content-loader";

export const PostsListLoader = () => (
  <ContentLoader viewBox="0 0 380 280">
    <rect x="0" y="0" rx="10" ry="5" width="110" height="130" />
    <rect x="130" y="0" rx="10" ry="5" width="110" height="130" />
    <rect x="260" y="0" rx="10" ry="5" width="110" height="130" />
    <rect x="0" y="140" rx="10" ry="5" width="110" height="130" />
    <rect x="130" y="140" rx="10" ry="5" width="110" height="130" />
    <rect x="260" y="140" rx="10" ry="5" width="110" height="130" />
  </ContentLoader>
);
