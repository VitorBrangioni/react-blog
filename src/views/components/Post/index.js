import React from "react";
import { useHistory } from "react-router-dom";
import DateHelper from "../../../helpers/DateHelper";
import {
  Container,
  Box,
  ContainerTitle,
  ContainerBody,
  Button,
  Title,
  Body,
  Legend,
} from "./styles";

const Post = ({ post }) => {
  const history = useHistory();
  const { title, body, metadata } = post;
  const { authorId, publishedAt } = metadata;

  return (
    <Container>
      <Box>
        <ContainerTitle>
          <Title>{title}</Title>
        </ContainerTitle>
        <ContainerBody>
          <Body>{`${body.slice(0, 200)}...`}</Body>
          <Legend>
            Author: Vitor Brangioni. | Data:{" "}
            {DateHelper.formatDatetime(publishedAt)}
          </Legend>
        </ContainerBody>

        <Button onClick={() => history.push("/post", { post })}>
          Ler mais
        </Button>
      </Box>
    </Container>
  );
};

export default Post;
