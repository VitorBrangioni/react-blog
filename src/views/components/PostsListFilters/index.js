import React, { Component } from "react";
import { Container } from "./styles";
import { connect } from "react-redux";
import { bindActionCreators } from "redux";
import { Creators as PostsActions } from "../../../store/ducks/postDuck";
import Select from "react-select";
const defaultAuthors = [{ value: null, label: "Todos" }];

class PostsListFilter extends Component {
  constructor() {
    super();
    this.state = {
      selectedAuthor: defaultAuthors[0],
      selectedOrder: { value: 1, label: "Crescente" },
      orders: [
        { value: -1, label: "Crescente" },
        { value: 1, label: "Decrescente" },
      ],
    };
  }

  handleAuthor(selectedAuthor) {
    this.props.getPosts(selectedAuthor.value, this.state.selectedOrder.value);
    this.setState({ selectedAuthor }, () =>
      console.log(`Option selected:`, this.state.selectedAuthor)
    );
  }

  handleOrder(selectedOrder) {
    this.props.getPosts(this.state.selectedAuthor.value, selectedOrder.value);
    this.setState({ selectedOrder }, () =>
      console.log(`Option selected:`, this.state.selectedOrder)
    );
  }

  getAuthorsForSelect() {
    const { authors } = this.props.resGetPosts;
    const authorsFormatted = authors.map(({ id, name }) => ({
      value: id,
      label: name,
    }));
    return [...defaultAuthors, ...authorsFormatted];
  }

  render() {
    const { selectedAuthor, selectedOrder, orders } = this.state;
    const authors = this.getAuthorsForSelect();

    return (
      <Container>
        <div>
          <h2>Autor:</h2>
        </div>
        <div>
          <Select
            value={selectedAuthor}
            onChange={this.handleAuthor.bind(this)}
            options={authors}
          />
        </div>
        <div>
          <h2>Ordem :</h2>
        </div>
        <div>
          <Select
            value={selectedOrder}
            onChange={this.handleOrder.bind(this)}
            options={orders}
          />
        </div>
      </Container>
    );
  }
}

const mapStateToProps = ({ PostDuck }) => {
  return {
    resGetPosts: PostDuck,
  };
};

const mapDispatchToProps = (dispatch) =>
  bindActionCreators(PostsActions, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(PostsListFilter);
