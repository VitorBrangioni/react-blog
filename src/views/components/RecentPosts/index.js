import React, { Component } from "react";
import { Container, Link } from "./styles";
import { connect } from "react-redux";
import { bindActionCreators } from "redux";
import { Creators as PostsActions } from "../../../store/ducks/postDuck";

class RecentPosts extends Component {
  render() {
    const { history } = this.props;
    const { recentPosts } = this.props.resGetPosts;

    return (
      <Container>
        <div>
          <h2>Recent Posts:</h2>
        </div>
        <div>
          {recentPosts.map((post, i) => (
            <Link
              key={`recent-post-${i}`}
              onClick={() => history.push("post", { post })}
            >
              <h3>- {post.title}</h3>
            </Link>
          ))}
        </div>
      </Container>
    );
  }
}

const mapStateToProps = ({ PostDuck }) => {
  return {
    resGetPosts: PostDuck,
  };
};

const mapDispatchToProps = (dispatch) =>
  bindActionCreators(PostsActions, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(RecentPosts);
