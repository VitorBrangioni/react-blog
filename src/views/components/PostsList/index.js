import React, { Component } from "react";
import Post from "../Post";
import { Container } from "./styles";
import { connect } from "react-redux";
import { bindActionCreators } from "redux";
import { Creators as PostsActions } from "../../../store/ducks/postDuck";
import { PostsListLoader } from "../Loader";

class PostsList extends Component {
  constructor(props) {
    super(props);
    this.props.getPosts(null, -1);
  }

  render() {
    const { data, loading } = this.props.resGetPosts;
    if (loading) {
      return (<PostsListLoader />);
    } 
    return (
      <Container>
        {data.map((post, i) => (
          <Post post={post} key={i} />
        ))}
      </Container>
    );
  }
}

const mapStateToProps = ({ PostDuck }) => {
  return {
    resGetPosts: PostDuck,
  };
};

const mapDispatchToProps = (dispatch) =>
  bindActionCreators(PostsActions, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(PostsList);
