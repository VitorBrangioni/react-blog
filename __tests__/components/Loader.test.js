import React from "react";
import { PostsListLoader } from "../../src/views/components/Loader";
import { shallow } from "enzyme"

describe("Loaders", () => {
  it("PostsListLoader snapshot", () => {
    const wrapper = shallow(<PostsListLoader />);

    expect(wrapper).toMatchSnapshot();
  });
});
