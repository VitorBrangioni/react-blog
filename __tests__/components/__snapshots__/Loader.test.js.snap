// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loaders PostsListLoader snapshot 1`] = `
<ContentLoader
  viewBox="0 0 380 280"
>
  <rect
    height="130"
    rx="10"
    ry="5"
    width="110"
    x="0"
    y="0"
  />
  <rect
    height="130"
    rx="10"
    ry="5"
    width="110"
    x="130"
    y="0"
  />
  <rect
    height="130"
    rx="10"
    ry="5"
    width="110"
    x="260"
    y="0"
  />
  <rect
    height="130"
    rx="10"
    ry="5"
    width="110"
    x="0"
    y="140"
  />
  <rect
    height="130"
    rx="10"
    ry="5"
    width="110"
    x="130"
    y="140"
  />
  <rect
    height="130"
    rx="10"
    ry="5"
    width="110"
    x="260"
    y="140"
  />
</ContentLoader>
`;
